services/src/main/scala/cromwell/pipeline/service/GitLabProjectVersioning.scala

```diff
@@ -148,9 +148,33 @@ class GitLabProjectVersioning(httpClient: HttpClient, config: GitLabConfig)
     implicit ec: ExecutionContext
   ): AsyncResult[List[GitLabVersion]] = ???
 
-  override def getFileTree(project: Project, version: Option[PipelineVersion])(
+  override def getFilesTree(project: Project, version: Option[PipelineVersion])(
     implicit ec: ExecutionContext
-  ): AsyncResult[List[String]] = ???
+  ): AsyncResult[Seq[FileTree]] = {
+
+    val versionId: String = version match {
+      case Some(version) => version.name
+      case None          => config.defaultFileVersion
```

Comment:
I guess it's not how it should work.
`defaultFileVersion` actually means `initialFileVersion`... (could you please rename it?)
So it's a version which we're using **if there're no versions in git.** 
In this case we're trying to get version from git, that means that **version already exists**. 

The correct behaviour here if version is `None` is to make a request to git and get the latest version. And if there is no version in git we should return error.

Reply:
I did the changes according to your comments. Could you skim and see if i omitted something, please?